nss/integration-tests/integration_test.go

```diff
@@ -81,18 +103,28 @@ func TestIntegration(t *testing.T) {
 			t.Parallel()
 
 			if tc.cacheDB == "" {
-				tc.cacheDB = "multiple_users_and_groups"
+				tc.cacheDB = defaultDbState
 			}
 
 			var socketPath string
-			var daemonStopped chan struct{}
 			if !tc.noDaemon && !tc.noCustomSocket {
-				ctx, cancel := context.WithCancel(context.Background())
-				socketPath, daemonStopped = testutils.RunDaemon(ctx, t, daemonPath, testutils.WithPreviousDBState(tc.cacheDB))
-				t.Cleanup(func() {
-					cancel()
-					<-daemonStopped
-				})
+				socketPath = defaultSocket
+				if tc.cacheDB != defaultDbState {
```

Comment:
in the default state, we don’t run the daemon then? Shouldn’t the code always run the daemon with the gpasswd mock to make it less special-cased?

Reply:
We always run the daemon. We have a default daemon running in the start of the test and then, for special cases that require a different db state, we run a separate instance.

Reply:
ack then!